lib/catalog/data_driven_form_validator.rb

```diff
@@ -0,0 +1,86 @@
+module Catalog
+  class DataDrivenFormValidator
+    COMPONENTS = %i[text-field
+                    textarea-field
+                    field-array
+                    select-field
+                    checkbox
+                    sub-form
+                    radio
+                    tabs
+                    tab-item
+                    date-picker
+                    time-picker
+                    wizard
+                    switch-field
+                    plain-text].freeze
+
+    VALIDATORS = %i[required-validator
+                    min-length-validator
+                    max-length-validator
+                    exact-length-validator
+                    min-items-validator
+                    min-number-value
+                    max-number-value
+                    pattern-validator
+                    url-validator].freeze
+
+    DATA_TYPES = %i[integer
+                    float
+                    number
+                    boolean
+                    string].freeze
+
+    def self.valid?(ddf)
+      ddf = ddf.class == Hash ? ddf.with_indifferent_access : JSON.parse(ddf).with_indifferent_access
+      fields?(ddf[:schema][:fields])
+
+      true
+    end
+
+    class << self
+      def fields?(fields)
+        fields.each do |field|
+          component?(field[:component].to_sym)
+          data_type?(field[:dataType].to_sym) if field.key?(:dataType)
+          validators?(field[:validate]) if field.key?(:validate)
+          options?(fields)
+        end
+      end
+
+      def component?(schema_component)
+        raise Catalog::InvalidSurvey unless COMPONENTS.include?(schema_component)
```

Comment:
@lindgrenj6 Should we have appropriate messages with each of these exceptions to say why we raised the InvalidSurvey. In the above case it would be component missing